### PR TITLE
fix: update ed25519 dep

### PR DESCRIPTION
**Pull request by chris13524**
# Description

Updates [WalletConnectRust dep](https://github.com/WalletConnect/WalletConnectRust/pull/64) to avoid 2 versions of ed25519 dependency. Remove unnecessary dependencies because these come from WalletConnectRust now.

Remaining work:
- [x] Merge https://github.com/WalletConnect/WalletConnectRust/pull/64 and use tagged version

## How Has This Been Tested?

Existing tests

## Due Diligence

* [ ] Breaking change
* [ ] Requires a documentation update
* [ ] Requires a e2e/integration test update

### CODE REVIEW

File: Cargo.toml [hunk 1]
```diff
@@ -74,8 +72,8 @@ futures = "0.3.26"
 futures-util = "0.3"
 dashmap = "5.4.0"
 
-relay_rpc = { git = "https://github.com/WalletConnect/WalletConnectRust.git", tag = "v0.26.2", features = ["cacao"] }
-relay_client = { git = "https://github.com/WalletConnect/WalletConnectRust.git", tag = "v0.26.2" }
+relay_rpc = { git = "https://github.com/WalletConnect/WalletConnectRust.git", branch = "53-upgrade-ed25519-dalek-dependency-to-200", features = ["cacao"] }
```
Comment (nopestack) [inline Cargo.toml:75]: will the contents of these branches be merged into main anytime soon?

**Review by nopestack — APPROVED** [review]
LGTM but hold off on merging if the branches on the upstream repo can be merged to main before merging this one so we don't need to set `branch` on these deps

**Review by geekbrother — APPROVED** [review]

File: Cargo.toml (hunk 1 above)
Comment (chris13524) [reply Cargo.toml:75]: Yes see PR description

Comment (chris13524) [reply Cargo.toml:75]: Using branch so it passes CI and will update to tagged version before merging

**Pull request merged**